rotkehlchen/api/rest.py

```diff
@@ -3036,9 +3036,15 @@ def decode_evm_transactions(
             chain_manager.transactions.get_receipts_for_transactions_missing_them()
             amount_of_tx_to_decode = dbevmtx.count_hashes_not_decoded(chain_id=evm_chain)
             if amount_of_tx_to_decode > 0:
-                chain_manager.transactions_decoder.get_and_decode_undecoded_transactions(
-                    send_ws_notifications=True,
-                )
+                # disallow scheduling of new tasks while decoding transactions
+                self.rotkehlchen.task_manager.should_schedule = False  # type: ignore  # task manager is not none here
```

Comment:
But this is not enough here. This is only if you ask for it directly. In every upgrade and migration this is done from the periodic task. It needs to happen there too. So better abstract it differently